js/app.js

```diff
@@ -0,0 +1,189 @@
+        this.ctx = this.canvasDOM.getContext('2d') // LINEA OBLIGATORIA para que sepa que es en 2d
+    },
+    setDimensions() {
+        this.canvasSize.w = 500 // esta ya definida arriba
+        this.canvasSize.h = 700 // esta ya definida arriba
+        this.canvasDOM.setAttribute('width', this.canvasSize.w) //establezco ancho y alto
+        this.canvasDOM.setAttribute('height', this.canvasSize.h) // LINEA OBLIGATORIA JUNTO CON LA DE ARRIBA
+    },
+    drawRoad() {
+        //console.log("funciono")
+        //console.log(this.canvasSize)
+        //void ctx.fillRect(x, y, width, height);
+        this.ctx.fillStyle = "green"// realizo un rectangulo verde que ocupe todo el canvasSize
+        this.ctx.fillRect(0, 0, this.canvasSize.w, this.canvasSize.h)
+        this.ctx.fillStyle = "grey"
+        this.ctx.fillRect(30, 20, this.canvasSize.w - 60, this.canvasSize.h - 50)
+
+        //LINEAS BLANCAS LATERALES
+        this.ctx.strokeStyle = 'white'
+        this.ctx.lineWidth = 6
+        this.ctx.beginPath() // Esta función sirve para decirle al contexto del canvas que vamos a empezar a dibujar un camino
+        this.ctx.moveTo(10, 0) //this.ctx.moveTo(posicion x, posicion y) INICIAL DE LA LINEA
+        //this.ctx.lineTo(posicion x, posicion y) FINAL DE LA LINEA DE LA LINEA
+        this.ctx.lineTo(10, 700)
+        this.ctx.stroke() // DIBUJA EL TRAZO DEL PUNTO INICIAL AL FINAL
+        //this.ctx.closePath() //
+
+        this.ctx.strokeStyle = 'white'
+        this.ctx.lineWidth = 6
+        this.ctx.beginPath()
+        this.ctx.moveTo(490, 0) // posicion inicial x, y
+        this.ctx.lineTo(490, 700) // posicion final x, y
+        this.ctx.stroke()
+        //this.ctx.closePath() //NO necesaria
+
+        //LINEA CENTRAL DISCONTINUA
+
+        this.ctx.lineWidth = 10
+        this.ctx.beginPath()
+        this.ctx.setLineDash([60, 30])
+        this.ctx.moveTo(this.canvasSize.w / 2, 20)
+        this.ctx.lineTo(this.canvasSize.w / 2, this.canvasSize.h - 30)
+        this.ctx.stroke()
+        // this.ctx.closePath()
+    },
+    insertImage() {
+        const imageInstance = new Image()
+        imageInstance.src = "./images/car.png"
+        //void ctx.drawImage(image, dx, dy, dWidth, dHeight);
+        imageInstance.onload = () => this.ctx.drawImage(imageInstance,
+            this.canvasSize.w / 2 - 40, this.canvasSize.h - 100, 80, 100)
+    },
+    clearScreen() {
+        this.ctx.clearRect(0, 0, this.canvasSize.w, this.canvasSize.h)
+    },
+    drawObstacle() {
+        const obs1 = new Obs(this.ctx, 30, 5, 200, 15, 1, this.canvasSize)
+        const obs2 = new Obs(this.ctx, 150, 180, 150, 30, 1.3, this.canvasSize)
+        const obs3 = new Obs(this.ctx, 20, 350, 160, 30, 1.8, this.canvasSize)
+        const obs4 = new Obs(this.ctx, 145, 340, 220, 30, 2, this.canvasSize)
+        const obs5 = new Obs(this.ctx, 200, 200, 200, 15, 1.45, this.canvasSize)
+    },

```

Comment:
Fijate que aquí te guardas los obstaculos en constantes y luego tratas de pintar a través de this.obstacle.

Esto con el proyecto seguro que ya no te está pasando :wink: